### PR TITLE
Add Liquid tag snippet completion

### LINKED ISSUE #500: Liquid tag end auto-insert

### PR DESCRIPTION
**Pull request by charlespwd**
## What are you adding in this PR?

- This PR adds snippet completion for Liquid tags. 
    - Accept the completion item for `if` and get `{% if condition %}\n  \n{% endif %}` with tabulated placeholders
    - Accept the completion item for `render` and get `{% render 'snippet' %}`
    - The snippets are smart depending on context:
       - Will infer whitespace stripping characters based on context
       - Will not snippet complete `{%` and `%}` if inside a `{% liquid %}` tag
       - Will not snippet complete if markup is already present in the tag definition

https://github.com/user-attachments/assets/70d6903b-5e9f-4a53-856d-bd1c806b56af

- I have verified that this works well in CodeMirror without changes :) 

https://github.com/user-attachments/assets/f1f9f14a-00af-4497-a5da-a616304b3f24

Fixes #500

## What did you learn?

Shit is hard. Lots of edge cases to cover. 

## Before you deploy
- [x] I have added a minor changeset

### CODE REVIEW

File: packages/theme-language-server-common/src/completions/providers/LiquidTagsCompletionProvider.spec.ts [hunk 1]
```diff
@@ -1,12 +1,80 @@
...
+    \$3
+  {% else %}
+    \$4
+{% endcase -%}`;
+const caseSnippetWithAllTheStripping = `case \${1:variable} -%}
+  {%- when \${2:first_value} -%}
+    \$3
+  {%- else -%}
+    \$4
+{%- endcase -%}`;
```
[the remark this answers is not in the document]
Comment (aswamy) [reply packages/theme-language-server-common/src/completions/providers/LiquidTagsCompletionProvider.spec.ts:32]: It only does it for `case` statements, but not `if`

File: packages/theme-language-server-common/src/completions/providers/LiquidTagsCompletionProvider.ts [hunk 2]
```diff
@@ -150,3 +160,299 @@ function findParentNode(partial: string, ancestors: LiquidHtmlNode[]): LiquidTag
...
+    .slice(textDocument.offsetAt(params.position), textDocument.offsetAt(endOfBlockStart))
+    .replace(/-?%\}/, '');
+}
+
+// We're trying to infer if we should trim the whitespace to the right given what the user has already written
+// {%  if|        => ''
+// {%- if|        => '-'
+// {%- if|  %}    => ''
+// {%- if| -%}    => '-'
+// {%  if| -%}    => '-'
```
[the remark this answers is not in the document]
Comment (charlespwd) [reply packages/theme-language-server-common/src/completions/providers/LiquidTagsCompletionProvider.ts:431]: Not a duplicate. The extra line is to show that if there's whitespace stripping to the _right_ in the original document, it won't try to copy the lack-of-whitespace-stripping shown on the left.

File: packages/theme-language-server-common/src/completions/providers/LiquidTagsCompletionProvider.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (charlespwd) [reply packages/theme-language-server-common/src/completions/providers/LiquidTagsCompletionProvider.spec.ts:32]: Oh but yeah, the default behaviour is that $0 is the last tab stop and that if ommitted from the snippet it is at the end of the string: 

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#:~:text=whereas%20%240%20denotes%20the%20final%20cursor%20position.%20Multiple%20tab%20stops%20are%20linked%20and%20updated%20in%20sync.

**Review by aswamy — APPROVED** [review]
Took a while to tophat all the edgecases, but i think we're good to go! Thanks for being so thorough with your test cases ❤️.

- We already talked about the edge case where you open a liquid tag inside an enclosed string (we have to manually trigger auto-complete suggestion for some reason). Seems like that's a VScode issue, but nothing we can really do about that.
- Just finish up [this comment](https://github.com/Shopify/theme-tools/pull/507#discussion_r1786405553), and we should be golden
